webpack/assets/javascripts/react_app/components/PF4/TableIndexPage/TableIndexPage.js

```diff
@@ -0,0 +1,251 @@
+import { getResults } from '../../AutoComplete/AutoCompleteActions';
+import { TRIGGERS } from '../../AutoComplete/AutoCompleteConstants';
+import BreadcrumbBar from '../../BreadcrumbBar';
+import SearchBar from '../../SearchBar';
+import Head from '../../Head';
+import { ActionButtons } from './ActionButtons';
+import './TableIndexPage.scss';
+
+const TableIndexPage = ({
+  apiUrl,
+  apiOptions,
+  header,
+  breadcrumbOptions,
+  beforeToolbarComponent,
+  controller,
+  searchable,
+  exportable,
+  creatable,
+  hasHelpPage,
+  customSearchProps,
+  customExportURL,
+  customCreateURL,
+  customHelpURL,
+  customActionButtons,
+  cutsomToolbarItems,
+  children,
+}) => {
+  const [params, setParams] = useState({});
+  // Adding to the deps apiOptions creates an endless loop
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const memoOptions = useMemo(() => ({ ...apiOptions, params }), [params]);
+  const history = useHistory();
+  const { location: { search } = {} } = history || {};
+  const {
+    response: { search: apiSearchQuery, can_create: canCreate },
+    status = STATUS.PENDING,
+  } = useAPI('get', apiUrl, memoOptions);
+
+  const dispatch = useDispatch();
+  const memoDefaultSearchProps = useMemo(
+    () => getControllerSearchProps(controller),
+    [controller]
+  );
+  const searchProps = customSearchProps || memoDefaultSearchProps;
+
+  const setAutocomplete = useCallback(
+    newSearch => {
+      dispatch(
+        getResults({
+          url: searchProps.autocomplete.url,
+          searchQuery: newSearch,
+          controller: searchProps.controller,
+          trigger: TRIGGERS.INPUT_CHANGE,
+          id: searchProps.autocomplete.id,
+        })
+      );
+    },
+    [searchProps, dispatch]
+  );
+  useEffect(() => {
+    let newSearch;
+    if (search !== undefined) {
+      const urlParams = new URLSearchParams(search);
+      newSearch = urlParams.get('search');
+    } else {
+      newSearch = getURIsearch();
+    }
+    setAutocomplete(newSearch || '');
+    setParams({ search: newSearch });
+  }, [search, apiSearchQuery, setAutocomplete]);
+
+  const setSearch = newSearch => {
+    if (history) {
+      const uri = new URI();
+      uri.setSearch(newSearch);
+      history.push({ search: uri.search() });
+    } else {
+      changeQuery(params);
+    }
+    setParams(newSearch);
+  };
+  const onBookmarkClick = newSearch => {
+    setAutocomplete(newSearch);
+    setSearch({ search: newSearch, page: 1 });
+  };
+
+  const actionButtons = [
+    creatable &&
+      canCreate && {
+        title: __('Create new'),
+        action: { href: customCreateURL || createURL() },
+      },
+    exportable && {
+      title: __('Export'),
+      action: { href: customExportURL || exportURL() },
+    },
+    hasHelpPage && {
+      title: __('Documentation'),
+      icon: <QuestionCircleIcon />,
+      action: { href: customHelpURL || helpURL() },
+    },
+    ...customActionButtons,
```

Comment:
should it be key based so we could override the action items? e.g:
```js
{
  create: { ... },
  export: { ... },
  help: { ... },
  ...customActionButtons,
}
```

Reply:
Devs can just pass `creatable={false}` and `exportable`,`hasHelpPage` are false by default so they wont be shown